vyper/codegen/expr.py

```diff
@@ -193,9 +193,8 @@ def parse_Name(self):
                 # using the folded value metadata
                 assert isinstance(varinfo.typ, StructT)
                 value_node = varinfo.decl_node.value
-                value_node = (
-                    value_node.get_folded_value() if value_node.has_folded_value else value_node
-                )
+                if value_node.has_folded_value:
+                    value_node = value_node.get_folded_value()
```

Comment:
i don't think this line is necessary, because it's handled when the Expr constructor gets called in the recursion